images/build/GitHubActions.Dockerfile

```diff
@@ -0,0 +1,165 @@
+ && . ${BUILD_DIR}/__nodeVersions.sh \
+ && ln -s $YARN_VERSION /opt/yarn/stable \
+ && ln -s $YARN_VERSION /opt/yarn/latest \
+ && ln -s $YARN_VERSION /opt/yarn/$YARN_MINOR_VERSION \
+ && ln -s $YARN_MINOR_VERSION /opt/yarn/$YARN_MAJOR_VERSION
+RUN set -ex \
+ && mkdir -p /links \
+ && cp -s /opt/yarn/stable/bin/yarn /opt/yarn/stable/bin/yarnpkg /links
+
+# This stage is used only when building locally
+FROM mcr.microsoft.com/dotnet/core/sdk:2.1 as buildscriptbuilder
+ARG BUILD_DIR
+ARG IMAGES_DIR
+COPY src/BuildScriptGenerator /usr/oryx/src/BuildScriptGenerator
+COPY src/BuildScriptGeneratorCli /usr/oryx/src/BuildScriptGeneratorCli
+COPY src/Common /usr/oryx/src/Common
+COPY build/FinalPublicKey.snk usr/oryx/build/
+COPY src/CommonFiles /usr/oryx/src/CommonFiles
+# This statement copies signed oryx binaries from during agent build.
+# For local/dev contents of blank/empty directory named binaries are getting copied
+COPY binaries /opt/buildscriptgen/
+WORKDIR /usr/oryx/src
+ARG GIT_COMMIT=unspecified
+ARG AGENTBUILD=${AGENTBUILD}
+ARG BUILD_NUMBER=unspecified
+ARG RELEASE_TAG_NAME=unspecified
+ENV GIT_COMMIT=${GIT_COMMIT}
+ENV BUILD_NUMBER=${BUILD_NUMBER}
+ENV RELEASE_TAG_NAME=${RELEASE_TAG_NAME}
+ARG AGENTBUILD=${AGENTBUILD}
+RUN if [ -z "$AGENTBUILD" ]; then \
+        dotnet publish -r linux-x64 -o /opt/buildscriptgen/ -c Release BuildScriptGeneratorCli/BuildScriptGeneratorCli.csproj; \
+    fi
+RUN chmod a+x /opt/buildscriptgen/GenerateBuildScript
+
+FROM main AS final
+ARG BUILD_DIR
+ARG IMAGES_DIR
+ARG SDK_STORAGE_ENV_NAME
+ARG SDK_STORAGE_BASE_URL_VALUE
+WORKDIR /
+
+ENV PATH="$PATH:/opt/oryx:/opt/yarn/stable/bin:/opt/hugo"
+COPY images/build/benv.sh /opt/oryx/benv
+RUN chmod +x /opt/oryx/benv
+RUN mkdir -p /usr/local/share/pip-cache/lib
+RUN chmod -R 777 /usr/local/share/pip-cache
+
+# .NET Core related environment variables
+ENV NUGET_XMLDOC_MODE=skip \
+	DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1 \
+	NUGET_PACKAGES=/var/nuget
+
+# Grant read-write permissions to the nuget folder so that dotnet restore
+# can write into it.
+RUN mkdir -p /var/nuget
+RUN chmod a+rw /var/nuget
+
+# Copy Yarn and Hugo related content
+COPY --from=nodetools-install /opt /opt
+
+# Build script generator content. Docker doesn't support variables in --from
+# so we are building an extra stage to copy binaries from correct build stage
+COPY --from=buildscriptbuilder /opt/buildscriptgen/ /opt/buildscriptgen/
+RUN ln -s /opt/buildscriptgen/GenerateBuildScript /opt/oryx/oryx
+
+RUN rm -rf /tmp/oryx
+
+# Bake Application Insights key from pipeline variable into final image
+ARG AI_KEY
+ENV ORYX_AI_INSTRUMENTATION_KEY=${AI_KEY}
+ENV ${SDK_STORAGE_ENV_NAME} ${SDK_STORAGE_BASE_URL_VALUE}
+
+ARG GIT_COMMIT=unspecified
+ARG BUILD_NUMBER=unspecified
+ARG RELEASE_TAG_NAME=unspecified
+LABEL com.microsoft.oryx.git-commit=${GIT_COMMIT}
+LABEL com.microsoft.oryx.build-number=${BUILD_NUMBER}
+LABEL com.microsoft.oryx.release-tag-name=${RELEASE_TAG_NAME}
+
+ENTRYPOINT [ "benv" ]
```

Comment:
nit: This dockerfile seems to have lot of similarities with azurefunction/slim.dockerfile of oryx ... can we add this as basic image layer to all the images?

Reply:
Check my earlier comment regarding this.